lib/ddtrace/sampler.rb

```diff
@@ -247,7 +248,12 @@ def priority_assigned_upstream?(span)
 
     def priority_sample!(span)
       preserving_sampling(span) do
-        @priority_sampler.sample!(span)
+        @priority_sampler.sample!(span).tap do
```

Comment:
There is a chance this isn't a `RateByServiceSampler`, should we add a check here?

e.g.

```
if @priority_sampler.is_a?(RateByServiceSampler) do
  span.set_metric(AGENT_RATE_METRIC_KEY, @priority_sampler.sample_rate(span))
end
```

Reply:
I think there were certain scenarios where this could be the case. I added the check accordingly.